src/CalculatorUITests/UnitConverterFunctionalTests.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using CalculatorUITestFramework;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium;
+using System;
+
+namespace CalculatorUITests
+{
+    [TestClass]
+    public class UnitConverterFunctionalTests
+    {
+        private static UnitConverterPage page = new UnitConverterPage();
+
+        /// <summary>
+        /// Initializes the WinAppDriver web driver session.
+        /// </summary>
+        /// <param name="context"></param>
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            // Create session to launch a Calculator window
+            WinAppDriver.Instance.SetupCalculatorSession(context);
+        }
+
+        /// <summary>
+        /// Closes the app and WinAppDriver web driver session.
+        /// </summary>
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            // Tear down Calculator session.
+            WinAppDriver.Instance.TearDownCalculatorSession();
+        }
+
+        /// <summary>
+        /// Ensures the calculator is in a cleared state
+        /// </summary>
+        [TestInitialize]
+        public void TestInit()
+        {
+            CalculatorApp.EnsureCalculatorHasFocus();
+            page.EnsureCalculatorIsCurrencyMode();
+            page.EnsureCalculatorResultTextIsZero();
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            page.ClearAll();
+        }
+
+        #region Basic UI Functionality via Mouse Input Tests
+
+        /// <summary>
+        /// These automated tests verify clicking each of the buttons in the Calculator UI and getting an expected result
+        /// Via mouse input, all basic UI functionality is checked 
+        /// </summary>
+        [TestMethod]
+        [Priority(0)]
+        public void MouseInput_EnterInputWithFullDecimalAndClear()
```

Comment:
I love that you've added UI tests to cover both basic functionality and your fix.  Also while we normally prefer unit tests over UI tests to cover functionality (as they tend to be both faster and more stable), this particular fix is tricky to unit test.

That said, because UI tests are harder to maintain, I'd like to try to keep these to as simple as possible.  In this case, I think it means having 3 tests:
1. a priority 0 test that covers the most basic of input.  ie: hits button 2 and then verifies both calculation results
2. a priority 1 test that covers backspace.  ie: inputs 2.13, verifies the initial input, and then backspaces all 3, checking the expected output each time.
3. a priority 1 test that covers clear.  ie: inputs 1.23, verifies the initial input, and then clicks clear, checking for the input to be cleared afterwards.

Thoughts?

Reply:
Sure I changed the UI tests no problem. I also added some unit test cases in addition to UI tests. The testing value however for the testing should be two digit value not one digit. So I kept "20.43" value in every test case.